### DISCUSSION TITLE
preload long sequence of images

### DISCUSSION THREAD

**Discussion by synapticlee** [jspsych/jsPsych#2935] [Q&A]
Hi, I have a collection of a dozen images, that I'd like to pregenerate a long sequence of trials from (N>1000). Ideally, I would read a csv with a list of numbers, that I would use to index a list of stimuli. How would I go about doing this?

**Comment by ChristopheBossens** [comment]
Do you have an example of the structure of your csv file? And would it be possible to perhaps create this sequence directly in JavaScript?

It is possible to read in a csv file, but in order for that to work your experiment needs to run on a webserver (which can be either on your own computer during testing and debugging, or on a remote server when your experiment goes live). After loading the csv file you will also have to parse it so that you get the information in a format that is proper for your experiment.

If you only need a list of numbers, what I typically do is create a small Python script that constructs this sequence, prints it as a string, and put that into a separate JavaScript file which I then read in by including that file into my main experiment. For example: the following would be the content of a script `experimental_images.js`:

```
let experimental_image_indices = [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 2, 4, 9]; // This list would contain all the indices
```

In the html file of my experiment, before the scipt tags for the jsPsych code, I would include:
```
<script src='experimental_images.js'>
<script>
console.log(experimental_image_indices ); // You now have access to your experimental images variable
const jsPsych = initJsPsych();
</script>
```

Is this something in the direction that you need?

**Reply by synapticlee** [reply]
Thanks! I could probably generate the sequence in javascript but I'm a little worried about the overhead with computing it on-line, or at the start of the experiment. I also know javaScript very little compared to other programming languages. 

I do only need a list of numbers, but this is a list that would be different per participant (and can't be read into jsPsych and arbitrarily randomized). I think now the easiest thing would be to have a folder of N (N = participants) .js files that are generated by e.g. python and read by jsPsych

What is console.log(experimental_image_indices ) ? How would it be different than calling a function called experimental_image_indices()?

**Reply by ChristopheBossens** [reply]
If you are worried about overhead, the best way is to test if it is actually a problem. In my experience, this is almost never actually a problem (but of course I do not know what needs to be computed). Could you perhaps try to explain how exactly the list should be created?

How do you plan to distribute the experiment (if each participant needs a different but predetermined list, you need a way of assigning a specific list to a specific participant)?

The console.log() function prints the content of the parameter you provide in the developers tool console window in your browser.

**Reply by synapticlee** [reply]
Thank you for your help! I was able to figure it all out by just writing my sequence generation in javascript.
